### ISSUE TITLE
Looking for maintainers

### ISSUE THREAD

**Issue by MathewSachin** [ManagedBass/ManagedBass#82]
The main use for ManagedBass for me was within a screen recording program I made. Eventually, it shifted to use NAudio to be completely open-source and because it had to run only on Windows. BASS is really great for multi-platform use.

But, I had many bad experiences (not related to BASS) in that screen recording program like people just blatantly copying it and selling it as their own on Microsoft Store and reporting those apps doesn't help. As a result, I've kind of grown frustrated of Windows and .NET as platforms to develop for and I'm shifting to other platforms like Android.

I would like to leave the project in the hands of people who continue to use it actively. Since ManagedBass is just a wrapper library, not much of an effort should be required for maintenance.
Particularly, I'd like if people who have already contributed to the repo could take up the ownership like @olitee @RCSDustinBond 
and the folks from osu!.

I don't really feel good leaving my projects like this, but, it's not much use to keep it stale when I'm not motivated enough to work on it 😢.

**Comment by RCSDustinBond** [comment]
Mathew,

I would be willing to oversee the project if you like, I do have a separate branch I maintain, with additional components like an AAC Encoder we paid for directly to IAN (Bass Developer). My company is in contact and does contract IAN for additions to his engine.

Dustin Bond
Senior Application Developer
RCS / IHeartMedia
www.rcsworks.com<http://www.rcsworks.com>
www.iheartmedia.com

**Comment by MathewSachin** [comment]
@RCSDustinBond That's really nice to hear. I have sent you an invite to add as a Member to the ManagedBass GitHub organization. After you accept, I'll convert you to an Owner.
I'll also share the NuGet keys by mail to you.

**Comment by MathewSachin** [comment]
@RCSDustinBond I think NuGet has changed how permissions work for organizations a bit. To give you access to the NuGet packages, I need to know your NuGet username.

**Comment by olitee** [comment]
Hi @MathewSachin, thanks for the callout. I think @RCSDustinBond is clearly better positioned to maintain this library day-to-day. My interest in this library is limited to a number of personal audio playback projects, that often need to sit on the back burner while I do the day job.

Happy to stay involved though.

**Comment by bpieslak** [comment]
@MathewSachin - sorry to hear about your bad experiences.  Thank you for making this project, and for all of the support you provided me in the early days.  I know I haven't been involved in a while, since my project basically reached steady-state and didn't require any further development or updates.  I know my project would not have been possible with out you and your efforts.  

Be well, and feel free to reach out if you ever need anything.
-Brian

**Comment by RCSDustinBond** [comment]
@MathewSachin - I created a new account on Nuget,  Dustin.Bond (dustinb@iheartmedia.com) ...

**Comment by FoggyFinder** [comment]
What's a status of this issue?

**Comment by d8ahazard** [comment]
@MathewSachin , @DustinBond  - Is this project still being maintained? Having some issues with a device, could really use a bit of assistance.

#86

**Comment by DustinBond** [comment]
Yes it is being maintained, I think is someone has some advice they will share it.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by mdsitton** [comment]
We also use ManagedBass for [clone hero](https://clonehero.net/) as well, so if needed could also potentially help out with maintenance.

I've just submitted a pr #87 for a bug i found today while converting some of our sound effects to use Opus.

**Comment by olitee** [comment]
Hi @DustinBond / @MathewSachin,

Do you have any plans for updating the nuget packages for this library? I'd be happy to assist in setting up an automated build/release process if you want?

I'm conscious that the current v3.0.0 nuget packages have some versioning issues, and have not been updated for almost 18 months - and that ParkSquare has recently forked this repo as 'BassSharp' (https://github.com/parksquare/BassSharp) to fix this issue.

While I'm tempted to switch to using the BassSharp nuget packages for my own projects, I feel it makes sense the solve the problem at the source - particularly since this repo has an established userbase and is relatively active.

Any thoughts? :)

**Comment by MathewSachin** [comment]
Automated releases would be great!
I can set up a NuGet key as a GitHub Action secret, or would you prefer collaborator access?

I personally didn't have any important projects depending on  ManagedBass/BASS now, so the packages were left stale...

**Comment by MathewSachin** [comment]
I've added a GitHub Actions secret: `NUGET_KEY` which can be used to push packages to NuGet.
This wouldn't be accessible from pull-requests due to security concerns.

**Comment by peppy** [comment]
As a heads up when releasing a new version: https://github.com/ManagedBass/ManagedBass/pull/93 may be regressing behaviour (we can't compile for iOS currently). Haven't investigated in detail yet.

**Comment by d8ahazard** [comment]
> Hi @DustinBond / @MathewSachin,
> 
> Do you have any plans for updating the nuget packages for this library? I'd be happy to assist in setting up an automated build/release process if you want?
> 
> I'm conscious that the current v3.0.0 nuget packages have some versioning issues, and have not been updated for almost 18 months - and that ParkSquare has recently forked this repo as 'BassSharp' (https://github.com/parksquare/BassSharp) to fix this issue.
> 
> While I'm tempted to switch to using the BassSharp nuget packages for my own projects, I feel it makes sense the solve the problem at the source - particularly since this repo has an established userbase and is relatively active.
> 
> Any thoughts? :)

Would it be possible to add package(s) that also include the native bass.dll and/or other required libraries, similar to how Emgu.CV packages the required openCV libraries for various operating systems? I think having the runtimes avaialable via NuGet as well would be incredibly useful.

**Comment by olitee** [comment]
@d8ahazard I can think of a number of pros and cons for this approach. The most obvious problem being how to intelligently handle switching between 32/64 bit versions of the libraries at compile time.

I'd also have to check with Un4Seen - as I'm pretty sure the licencing for those DLLs doesn't allow us to distribute them as part of a library. Their licence specifically states:

"Please note the products must be end-user products, eg. not components
used by other products. "

But I could reach out to Ian in the forums there for clarification.

**Comment by DustinBond** [comment]
Mathew, can we add Oli, to also help out. It would be great to get the nuget packages updated.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Oli Trenouth ***@***.***>
Sent: Saturday, August 7, 2021 11:40:43 AM
To: ManagedBass/ManagedBass ***@***.***>
Cc: Bond, Dustin ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [ManagedBass/ManagedBass] Looking for maintainers (#82)


@d8ahazard<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fd8ahazard&data=04%7C01%7Cdustinb%40iheartmedia.com%7Cac4703d95ca846073fb308d959d2dd13%7C122a527e5b714eba878d9810b495b9e3%7C0%7C0%7C637639584459198072%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=tFA5WxA6KuB4jHPmbDp1CuIWkcD97072sbIn%2BCuxmFQ%3D&reserved=0> I can think of a number of pros and cons for this approach. The most obvious problem being how to intelligently handle switching between 32/64 bit versions of the libraries at compile time.

I'd also have to check with Un4Seen - as I'm pretty sure the licencing for those DLLs doesn't allow us to distribute them as part of a library. Their licence specifically states:

"Please note the products must be end-user products, eg. not components
used by other products. "

But I could reach out to Ian in the forums there for clarification.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FManagedBass%2FManagedBass%2Fissues%2F82%23issuecomment-894691525&data=04%7C01%7Cdustinb%40iheartmedia.com%7Cac4703d95ca846073fb308d959d2dd13%7C122a527e5b714eba878d9810b495b9e3%7C0%7C0%7C637639584459208022%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=kkjw%2BiKqw2Po8OwB0YQ4ovEn9mt3YbfpHugyYi2XlUE%3D&reserved=0>, or unsubscribe<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAIW3H6YI5K4GURRDZMCMPIDT3V42XANCNFSM4PKCSGGQ&data=04%7C01%7Cdustinb%40iheartmedia.com%7Cac4703d95ca846073fb308d959d2dd13%7C122a527e5b714eba878d9810b495b9e3%7C0%7C0%7C637639584459208022%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=4aF2Fj%2FV%2BoEJMFaeYbqoM%2Fa3ShG21ROu9Gh%2FqZ5c5zI%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cdustinb%40iheartmedia.com%7Cac4703d95ca846073fb308d959d2dd13%7C122a527e5b714eba878d9810b495b9e3%7C0%7C0%7C637639584459217991%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=f92CBmqB%2FMj%2BVRnBTrAJ%2FYipRXTUCnwj3d6NRH86p%2FU%3D&reserved=0> or Android<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26utm_campaign%3Dnotification-email&data=04%7C01%7Cdustinb%40iheartmedia.com%7Cac4703d95ca846073fb308d959d2dd13%7C122a527e5b714eba878d9810b495b9e3%7C0%7C0%7C637639584459217991%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=BfGSwqeEMrEpGErHmG7gwmi9j9LPfBzKSX%2FV3%2FQURz4%3D&reserved=0>.

CAUTION: This message originated from outside of the company. Please take all precautions when opening attachments or links from outside sources.

**Comment by MathewSachin** [comment]
I've sent him an invite 👍

**Comment by olitee** [comment]
@DustinBond @MathewSachin Thanks for the invite. 

I've setup a pair of github Actions: One to act as a build-test (replacing appveyor), and the other to build, package and publish the nuget packages. All of this is currently on `automation/github-action-for-package-publishing` branch for now, while I test.

I'm thinking the best way of handling the nuget packages will be via the repository's Releases feature. So **creating** a **release** with a tag using semantic versioning (e.g. v.3.0.1, etc) will trigger the action to create and publish the nuget packages.

This approach also supports prerelease versions (e.g. v3.0.1-prerelease, or v3.0.1-beta) which could be handy.

It gives us complete control over the releases - rather than simply firing out a new package when the master branch changes.

Are you both OK with that approach?

**Comment by MathewSachin** [comment]
Yes, sounds perfect to me.

**Comment by d8ahazard** [comment]
> @d8ahazard I can think of a number of pros and cons for this approach. The most obvious problem being how to intelligently handle switching between 32/64 bit versions of the libraries at compile time.
> 
> I'd also have to check with Un4Seen - as I'm pretty sure the licencing for those DLLs doesn't allow us to distribute them as part of a library. Their licence specifically states:
> 
> "Please note the products must be end-user products, eg. not components
> used by other products. "
> 
> But I could reach out to Ian in the forums there for clarification.

Yeah, that would be awesome. Heck, maybe Un4Seen would be open to creating nuget package(s) directly...which would alleviate any issues with licensing.

In regards to handling the binaries at runtime...I *think* the solution would be to just emulate what EMGU.CV is doing for their runtime libraries, which handles this exact situation:

https://github.com/emgucv/emgucv/blob/master/Emgu.CV.Runtime/Windows/Emgu.CV.Runtime.Windows.projitems

**Comment by olitee** [comment]
@d8ahazard I like this idea, but I think it needs opening up for investigation and discussion. Could you create a new issue for this, as it's a little lost in this maintainers thread.

**Comment by olitee** [comment]
For those interested, [v3.1.0](https://github.com/ManagedBass/ManagedBass/releases/tag/3.1.0) of the ManagedBass nuget packages have now been released:

https://www.nuget.org/packages/ManagedBass/3.1.0

**Comment by mysteryx93** [comment]
I'm not sure that including the unmanaged DLLs is a good idea, because you do not know which DLLs the application wants to include or not.

Here's my solution, I created a project with all the DLLs, and I simply reference the BassDlls project to get all the DLLs for the desired platform. There's another BassEncDlls project for when I need to encode.
https://github.com/mysteryx93/HanumanInstituteApps/blob/master/DLL/Bass/BassDlls.csproj

**Comment by Nebula-Developer** [comment]
> I created a project with all the DLLs, and I simply reference the BassDlls project to get all the DLLs for the desired platform.

If another project references a library that uses this copy method, Bass looks through the executing project's binary directory, not the library's (despite the call to `Bass.Init` being in the library's assembly)

Is there a way to solve this?

**Comment by DustinBond** [comment]
@olitee Please try sending me another direct email. I didn't get the email I sent you a few weeks ago. my email is on my profile, i was not able to find yours.

**Comment by HakanL** [comment]
I'm also open to help out, I already maintain several open-source projects and I use ManagedBass in a commercial product I've built.

**Comment by HakanL** [comment]
@olitee Yeah, I don't have any issues that need fixing, but I'm still happy to be another available participant for any issues and review PRs. I've upgraded to 4.01 and it works well (actually there is a crash that I'm investigating but it's too early to see if it's related to this wrapper, but it seems like it's a unmanaged crash).
I'm developing on Windows but the product runs on Raspberry Pi CM4/CM5 in Linux (via Balena/Docker).

**Comment by KushalLukhi** [comment]
Hi maintainers — I'm Kushal (@KushalLukhi). I'd like to volunteer as a maintainer/contributor for this project.

I can help with:
- issue triage and reproduction
- reviewing/fixing open bugs
- dependency/security updates
- release/support chores

Happy to start with PR-based contributions and earn trust progressively. If you're open, please share the preferred onboarding path and first tasks.

**Comment by mysteryx93** [comment]
What's the current status for maintainers? I have submitted a PR for a bug fix; wondering when it will get accepted.

**Comment by HakanL** [comment]
@mysteryx93 It doesn't seem like there's been any updates :(

**Comment by mysteryx93** [comment]
ok someone approved the changes although it's not merged yet.

If someone has a MacOS, it would be great if you could test it.

I wrote the first code version but it wasn't working; I rewrote this 2nd version. It should be closer to working but remains untested. It would be good to know for sure.

Remains to see how long before a new package can be published. Currently the only owner on NuGet is ManagedBass. There's one of my packages I'm no longer developing where I added a 2nd owner, this can be done.

**Comment by HakanL** [comment]
I approved because the changes look good (but Copilot found a few items that I think would be good to address, mostly in documentation). But I'm not a maintainer, so my "approval" doesn't really mean anything, it's just that I reviewed the changes.

**Comment by olitee** [comment]
@mysteryx93 Thanks for checking in - @DustinBond and I are still here, so please bear with us. @HakanL thanks for your assistance, let me check in with Dustin and see if we can get your access upgraded.

**Comment by DustinBond** [comment]
@olitee Looking good, just let me know when your done and I can take a l deeper look, if you like. I was also wanting to do a release on the .net 10 stuff... I had it in pre-release for now..
